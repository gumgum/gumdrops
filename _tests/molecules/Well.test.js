import renderer from 'react-test-renderer';
import React from 'react';
import Well from '../../components/molecules/Well';

const defaultProps = {
    text: 'Foo',
    context: 'primary',
    className: '',
    style: { width: 100 }
};

test('Expect <Well> to render properly', () => {
    const tree = renderer.create(<Well {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Expect <Well> to render with a button', () => {
    const props = {
        ...defaultProps,
        button: false,
        callback: jest.fn()
    };
    const tree = renderer.create(<Well {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
});
