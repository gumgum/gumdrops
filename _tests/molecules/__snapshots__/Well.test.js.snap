// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Well> to render properly 1`] = `
<div
  className="gds-well gds-well--primary"
  style={
    {
      "width": 100,
    }
  }
>
  <p
    className="gds-well__text"
  >
    Foo
  </p>
</div>
`;

exports[`Expect <Well> to render with a button 1`] = `
<div
  className="gds-well gds-well--primary"
  style={
    {
      "width": 100,
    }
  }
>
  <p
    className="gds-well__text"
  >
    Foo
  </p>
</div>
`;
